frmts/jpegxl/jpegxl.cpp

```diff
@@ -1001,231 +1010,269 @@ char **JPEGXLDataset::GetMetadata(const char *pszDomain)
+                    }
+                    else if (status == JXL_DEC_BASIC_INFO ||
+                             status == JXL_DEC_FULL_IMAGE)
+                    {
+                        // do nothing
+                    }
+                    else
                     {
-                        std::vector<GByte> abyNew;
-                        abyNew.reserve(jpeg_bytes.size() + 2 + nEXIFMarkerSize);
-                        abyNew.insert(abyNew.end(), jpeg_bytes.data(),
-                                      jpeg_bytes.data() + nInsertPos);
-                        abyNew.insert(abyNew.end(), static_cast<GByte>(0xFF));
-                        abyNew.insert(abyNew.end(), static_cast<GByte>(0xE1));
-                        abyNew.insert(abyNew.end(),
-                                      static_cast<GByte>(nEXIFMarkerSize >> 8));
-                        abyNew.insert(
-                            abyNew.end(),
-                            static_cast<GByte>(nEXIFMarkerSize & 0xFF));
-                        abyNew.insert(abyNew.end(), EXIF_SIGNATURE,
-                                      EXIF_SIGNATURE + sizeof(EXIF_SIGNATURE));
-                        abyNew.insert(abyNew.end(), m_abyEXIFBox.data(),
-                                      m_abyEXIFBox.data() +
-                                          m_abyEXIFBox.size());
-                        abyNew.insert(abyNew.end(),
-                                      jpeg_bytes.data() + nInsertPos,
-                                      jpeg_bytes.data() + jpeg_bytes.size());
-                        jpeg_bytes = std::move(abyNew);
+                        CPLError(CE_Warning, CPLE_AppDefined,
+                                 "Unexpected event: %d", status);
+                        break;
                     }
                 }
+                if (bJPEGReconstruction)
+                {
+                    size_t used_jpeg_output =
+                        jpeg_data_chunk.size() -
+                        JxlDecoderReleaseJPEGBuffer(decoder.get());
+                    jpeg_bytes.insert(jpeg_bytes.end(), jpeg_data_chunk.data(),
+                                      jpeg_data_chunk.data() +
+                                          used_jpeg_output);
+                }
 
-                constexpr const char APP1_XMP_SIGNATURE[] =
-                    "http://ns.adobe.com/xap/1.0/";
-                const size_t nXMPMarkerSize =
-                    2 + sizeof(APP1_XMP_SIGNATURE) + m_osXMP.size();
-                if (!m_osXMP.empty() && nXMPMarkerSize <= 65535U)
+                JxlDecoderReleaseInput(decoder.get());
+
+                if (!jpeg_bytes.empty() &&
+                    jpeg_bytes.size() < static_cast<size_t>(INT_MAX))
                 {
+                    constexpr GByte EXIF_SIGNATURE[] = {'E', 'x',  'i',
+                                                        'f', '\0', '\0'};
+                    constexpr const char APP1_XMP_SIGNATURE[] =
```

Comment:
```suggestion
                    constexpr char APP1_XMP_SIGNATURE[] =
```

Perhaps you can use `static const` instead of `constexpr` here (and in the previous occurrence).